plugins/idea/README.md

```diff
@@ -23,29 +28,38 @@ The output of this folder is as follows:
-To build the plugin, you need to have IntelliJ installed. From IntelliJ, simply browse to the location of this folder, and open it as an IntelliJ project. The project can be built by clicking on `Build -> Build Project`. Once this step is done, you should be able to run/debug the plugin.
 
-Note: You might need to tweak the IDE project settings to correctly configure the plugin project. The two following steps are required:
+This plugin can be built with `gradle`. To build the plugin, simply run the following command from the `idea` folder:
+
+`sh gradlew clean build`
+
+This will download the required IntelliJ dependencies, will build the plugin and will place the results in the `build/distributions` folder.
 
-* When the IDE project for the plugin is first opened, IntelliJ will complain about the lack of a path variable called `JTREG_HOME`. Please follow the IDE instructions, and set this path variable to point to the root of your jtreg build output or your jtreg installation (`JTREG_HOME/libs/javatest.jar` must exist). The setting can also be found in under `Settings -> Appearance and Behavior -> Path Variables`. This step is required as the project depends on some of the jtreg libraries.
+> Note: to build the plugin, the build script must point to a valid jtreg installation; see the property `jtregHome` in the `gradle.properties` file, and tweak accordingly.
 
-* You need to create an Intellij SDK and name it `IDEA JDK` for the project to work correctly. To create a new SDK, click on `File -> Project Structure`. When the dialog opens, click on the `SDKs` entry in the left pane, and add a new SDK (by clicking on the *plus* icon). The SDK we want to create is an 'IntelliJ Platform Plugin SDK'. Once the new SDK type has been selected, you will have to tell IntelliJ what's the SDK path - you will see that IntelliJ will already point to the folder containing your IntelliJ installation - that is a fine choice. Rename it to `IDEA JDK`. Secondly, you will have to tell IntelliJ which JDK to use together with the plugin SDK. Any JDK 8 (or greater) is a fine choice here.
-  
-Note: by running the plugin here we mean running it in a sandbox environment - for truly installing the plugin in your IDE, see the steps below.
+> Note: the property `intellijVersion` can be used to specify which IDE version should the plugin depend on (defaults to `2021.1`).
+
+Once the build is configured correctly, the plugin can even be tested in a sandbox environment, as follows:
+
+`sh gradlew runIde`
 
 ### Installing the plugin
 
-To install the plugin in your IDE, first you need to prepare a plugin module file (a `.jar` file). This can be done by selecting `Build -> Prepare Plugin Module for Deployment`. Once that is done, a new jar file should be created under the hidden `.idea` folder (and a related pop up notification should appear).
+To install the plugin in your IDE, first you need to build a plugin module file (a `.jar` file), as described in the previous section.
```

Comment:
Several comments here:

1.  The addition in the filename "-gradle" is a little confusing, is it really necessary?

2.  There are two jar files in my resulting build directory, you have to compare them to know they are the same.  Just adds a speedbump for someone trying for the first time.

./build/libs/jtreg-plugin-gradle-1.12.jar
./build/idea-sandbox/plugins/jtreg-plugin-gradle/lib/jtreg-plugin-gradle-1.12.jar

3.  IntelliJ can also install a .zip file, but you said to install the jar file.  Is it possible to not build the .zip file?

./build/distributions/jtreg-plugin-gradle-1.12.zip

4.  Lastly, I built the plugin with my default JDK 16 (60.0), but apparently Intellij IDEA 2021.1 only recognized up to JDK 11 (55.0) and hung after throwing up a popup.  You might want to -target 11?

Reply:
> Several comments here:
> 
>     1. The addition in the filename "-gradle" is a little confusing, is it really necessary?

I didn't do anything to add that - maybe it can be removed, I'll check
> 
>     2. There are two jar files in my resulting build directory, you have to compare them to know they are the same.  Just adds a speedbump for someone trying for the first time.

The README talks about `build/distributions`. In there there is just _one_ file:

build/distributions/jtreg-plugin-gradle-1.12.zip

That is the plugin that should be installed.

> 
> 
> ./build/libs/jtreg-plugin-gradle-1.12.jar
> ./build/idea-sandbox/plugins/jtreg-plugin-gradle/lib/jtreg-plugin-gradle-1.12.jar
> 
>     1. IntelliJ can also install a .zip file, but you said to install the jar file.  Is it possible to not build the .zip file?

The zip file is the one to install. I'll drop references to `jar` and replace with `zip`

> 
> 
> ./build/distributions/jtreg-plugin-gradle-1.12.zip
> 
>     1. Lastly, I built the plugin with my default JDK 16 (60.0), but apparently Intellij IDEA 2021.1 only recognized up to JDK 11 (55.0) and hung after throwing up a popup.  You might want to -target 11?

I'll check that.

Reply:
Missed the zip, I probably just read it as jar since you talked about building the jar in the para above.